CMakeLists.txt

```diff
@@ -253,9 +253,27 @@ rocm_package_setup_component(tests
 )
 
 add_subdirectory(library)
-add_subdirectory(example)
-add_subdirectory(test)
-add_subdirectory(profiler)
+
+option(BUILD_ALL, "Whether to build all targets, such as examples, tests, and profiler." OFF)
+option(WITH_EXAMPLES, "Whether to build examples." OFF)
+option(WITH_TESTS, "Whether to build tests." OFF)
+option(WITH_PROFILER, "Whether to build profiler." OFF)
```

Comment:
Can we place all cmake options in one place? There is already one for INT4 in this file some lines above.

Reply:
OK